.github/workflows/create-scheduled-release.yml

```diff
@@ -0,0 +1,78 @@
+name: "Create scheduled release"
+
+on:
+  schedule:
+    - cron: "0 0 1 * *"
+    - cron: "0 0 15 * *"
+  workflow_dispatch:
+
+permissions:
+  contents: write
+
+jobs:
+  get-changed-metadata:
+    name: "📋 Get a list of changed metadata"
+    runs-on: "ubuntu-20.04"
+    timeout-minutes: 5
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+      none-found: ${{ steps.set-matrix.outputs.none-found }}
+    steps:
+      - name: "☁️ Checkout repository"
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: "🔧 Setup java"
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'oracle'
+          java-version: '17'
+      - name: "🕸️ Get changed metadata matrix"
+        id: set-matrix
+        run: |
+          LATEST_TAG=$(ls -1 .git/refs/tags/ | sort -V | tail -1)
+          ./gradlew generateMatrixDiffCoordinates -PbaseCommit=$(git show-ref -s $LATEST_TAG) -PnewCommit=$(git rev-parse HEAD)
+
+  release:
+    needs: get-changed-metadata
+    if: needs.get-changed-metadata.result == 'success' && needs.get-changed-metadata.outputs.none-found != 'true'
+    name: "🚀 Create a release"
+    runs-on: "ubuntu-20.04"
+    env:
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - name: "☁️ Checkout repository"
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: "🔧 Setup java"
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'graalvm'
+          java-version: '17'
+      - name: "⬆️ Update version"
+        run: |
+          PREVIOUS_RELEASE_TAG=$(ls -1 .git/refs/tags/ | sort -V | tail -1)
+          echo "PREVIOUS_RELEASE_TAG=$PREVIOUS_RELEASE_TAG" >> ${GITHUB_ENV}
+          
+          CURRENT_RELEASE_TAG=$(sed -E 's/^([0-9]+\.)([0-9]+\.)([0-9]+)/echo \1\2$((\3+1))/e' <<< $PREVIOUS_RELEASE_TAG)
+          echo "CURRENT_RELEASE_TAG=$CURRENT_RELEASE_TAG" >> ${GITHUB_ENV}
+          
+          sed -i "s/project.version(\"0.3.10-SNAPSHOT\")/project.version(\"$CURRENT_RELEASE_TAG\")/g" build.gradle
```

Comment:
`0.3.10-SNAPSHOT` must be replaced with new, fixed version once we agree how it should look like.

From that point on, we will always have the same `SNAPSHOT` version on master, and concrete version in each "release commit".

Reply:
Is this the only place that needs to be updated? the tests won't need to be updated because they will need to match the version on the main branch, which currently is `0.3.10-SNAPSHOT`, right?

Reply:
Yes, this is the only place where we have to update project version.

Reply:
Version will be changed to `1.0.0-SNAPSHOT` in [this PR](https://github.com/oracle/graalvm-reachability-metadata/pull/541). Mentioned version will only act as a placeholder version that will be changed in actual releases (through CI)

Reply:
Why do it in a separate PR though?

Reply:
+1, let's do it in the same PR to make it easier

Reply:
Done!